yktracec/src/ykllvmwrap.cc

```diff
@@ -491,3 +491,56 @@ extern "C" void *__yktracec_irtrace_compile_for_tc_tests(
                         DebugInfoPath, nullptr, nullptr, 0);
 }
 #endif
+
+/// Represents an index and name of an LLVM IR function.
+extern "C" struct IRFunctionNameIndex {
+  unsigned int index;
+  const char *name;
+};
+
+/// Deallocates memory allocated for an array of `IRFunctionNameIndex`
+/// structures.
+extern "C" void free_function_names(IRFunctionNameIndex *function_names) {
+  delete[] function_names;
```

Comment:
Where is the `new` that allocated what you are `delete`ing here?

But before you answer that: I think `get_function_names` is computing a (conceptual) map from index to function name, right?

Assuming so, isn't this the kind of thing that we will want to use time and time again? And since the AOT module doesn't change, and is the same for all threads, I'd be happy for this map to be computed once on first use, and kept around for future tracing sessions to use. You can just let the death of the process be the freeing step :P

[`std::call_once`](https://en.cppreference.com/w/cpp/thread/call_once) may help.

Reply:
sounds good

Reply:
updated 👉 3a6d4b4386704ae8553725c55a9e5749c43717e2

Reply:
I moved `called_once` and function match process to `swt` module 👉 ee2a93c05910594526ea8c4a9c61de121916f2da